### PR TITLE
Play sound notification when pinged

### PR DESCRIPTION
**Pull request by ThePhilgrim**
This currently works on mac. I have no way of testing it on Windows/Linux.

@Arrinao Could you do a windows test maybe?

Should tick box 3 in #142

### CODE REVIEW

File: mantaray/views.py [hunk 1]
```diff
@@ -156,16 +146,15 @@ def add_notification(self, popup_text: str) -> None:
         self._update_view_selector()
         self.irc_widget.event_generate("<<NotificationCountChanged>>")
         _show_popup(self.view_name, popup_text)
+        playsound("mantaray/audio/notify.mp3")
```
Comment (Akuli) [inline mantaray/views.py:149]: Please include a setting in the setting dialog (right-click a server, "Server settings...") for disabling this. Maybe it shouldn't even be on by default? It doesn't have to be channel-specific IMO, just a checkbox that applies to all channels.

Ideally the same setting would apply to all servers, but there is no setting dialog for not-server-specific settings yet, and you don't have to create one just for this PR. (Mantaray supports being connected to multiple servers at once, but there's not yet a GUI for adding more servers.)

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,4 @@
 appdirs==1.4.4
 ttkthemes==3.2.2
+playsound==1.2.2
+PyObjC==8.1; sys_platform == "darwin"
```
Comment (Akuli) [inline requirements.txt:4]: Why is this needed? I think it deserves a short comment explaining that it has to do with `playsound`.

File: mantaray/views.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import sys
 import tkinter
 import subprocess
+from playsound import playsound  # type: ignore
```
Comment (Akuli) [inline mantaray/views.py:9]: Would you like to contribute stubs for this in typeshed? I can guide you through how that works, and also check whether our documentation about this is up to date. There are tools to do most of the work for you.

File: mantaray/views.py (hunk 1 above)
Comment (ThePhilgrim) [reply mantaray/views.py:160]: @Akuli I'm trying to figure out how to get the boolean from config to `class View`where this code is. However, `View` is not instantiated anywhere that I can see with `git grep`, so I cannot add any arguments to the instantiation. Nothing else in `View` seem to be using the config so I'm not sure how to do this.

File: mantaray/views.py (hunk 3 above)
Comment (ThePhilgrim) [reply mantaray/views.py:9]: Sure! Should be simple. `playsound` only has one single function :)

File: mantaray/views.py (hunk 1 above)
Comment (Akuli) [reply mantaray/views.py:160]: `View` itself is not instantiated, but its subclasses `ServerView` and `ChannelView` are. This confusion is one reason why I don't like inheritance in general, but I found it really makes sense for mantaray's views.

Because this is a server-specific setting, it should be accessed through the server view, e.g. `self.server_view.core.audio_notification`. However, the backend doesn't play audio, so it would make sense to just store the setting in the server view instead of getting the backend involved, as in `self.server_view.audio_notification`. Look at the `extra_notifications` setting for how this is done.

File: mantaray/views.py (hunk 3 above)
Comment (Akuli) [reply mantaray/views.py:9]: Our documentation seems to be terribly outdated. We have a new script that makes this kind of thing very easy, but it's broken with the latest mypy not documented. I made a pull request that fixes it. Once you have made the playsound stubs, would you like to make a pull request that also documents the process? :D

Here's how the process should work:
- Fork and clone typeshed. Make a new branch.
- Make a venv and `pip install -r requirements-tests.txt`
- Install playsound with pip
- Run `python3 scripts/create_baseline_stubs.py playsound`. Follow the instructions.

**Comment by ThePhilgrim** [comment]
https://user-images.githubusercontent.com/76887896/148661961-2b9d49a4-7545-4f16-9c8f-90185d38624e.mp4

File: mantaray/config.py [hunk 4]
```diff
@@ -245,6 +230,7 @@ def __init__(
         self._server_entry.var.set(initial_config["host"])
         self._ssl_var.set(initial_config["ssl"])  # must be before port
         self._port_entry.var.set(str(initial_config["port"]))
+        self._audio_var.set(initial_config["audio_notification"])
```
Comment (Akuli) [inline mantaray/config.py:233]: Nit: While a separate `.set()` works, it would be simpler to create `_audio_var` with `value=initial_config["audio_notification"]`.

**Comment by ThePhilgrim** [comment]
@Akuli Would you please black the code to fit the project settings?

File: mantaray/views.py [hunk 5]
```diff
@@ -155,6 +156,9 @@ def add_notification(self, popup_text: str) -> None:
         self.notification_count += 1
         self._update_view_selector()
         self.irc_widget.event_generate("<<NotificationCountChanged>>")
+        if self.server_view.audio_notification:
+            playsound("mantaray/audio/notify.mp3", False)
```
Comment (Akuli) [inline mantaray/views.py:160]: It would be good to wrap this in try/except, just in case playing the sound fails. Even if something unexpected goes wrong, the user should still see the popup and be able to just continue using mantaray.

In fact, this errors on linux, and the error prevents you from seeing the popup:

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.9/tkinter/__init__.py", line 1892, in __call__
    return self.func(*args)
  File "/usr/lib/python3.9/tkinter/__init__.py", line 814, in callit
    func(*args)
  File "/home/akuli/mantaray/mantaray/views.py", line 448, in handle_events
    channel_view.add_notification(f"<{event.sender}> {event.text}")
  File "/home/akuli/mantaray/mantaray/views.py", line 160, in add_notification
    playsound("mantaray/audio/notify.mp3", False)
  File "/home/akuli/mantaray/env/lib/python3.9/site-packages/playsound.py", line 80, in _playsoundNix
    raise NotImplementedError(
NotImplementedError: block=False cannot be used on this platform yet
```

This is my fault though. I wrote that `raise NotImplementedError(` code.

Comment (Akuli) [reply mantaray/views.py:160]: Turns out that this is fixed in playsound 1.3.0. Just change the version in requirements.txt.

File: mantaray/views.py [hunk 6]
```diff
@@ -155,6 +156,12 @@ def add_notification(self, popup_text: str) -> None:
         self.notification_count += 1
         self._update_view_selector()
         self.irc_widget.event_generate("<<NotificationCountChanged>>")
+        if self.server_view.audio_notification:
+            try:
+                playsound("mantaray/audio/notify.mp3", False)
+            except:
+                pass
```
Comment (Akuli) [inline mantaray/views.py:163]: - We don't want to catch literally all errors, e.g. `KeyboardInterrupt` which would make the program do nothing if you Ctrl+C while playing a sound.
- It would be good to print the full error message when an error occurs, so that you can debug why sound isn't playing.

```suggestion
            except Exception:
                traceback.print_exc()
```

**Comment by Akuli** [comment]
Now fix the test failures, and this is ready to be merged :)

**Review by Akuli — APPROVED** [review]
Thanks for working on mantaray!
